applications/RANSApplication/custom_strategies/algebraic_flux_corrected_steady_scalar_scheme.h

```diff
@@ -0,0 +1,582 @@
+        mrPeriodicIdVar(Variable<int>::StaticObject())
+    {
+        KRATOS_INFO("AlgebraicFluxCorrectedSteadyScalarScheme")
+            << " Using residual based algebraic flux corrected scheme with "
+               "relaxation "
+               "factor = "
+            << std::scientific << mRelaxationFactor << "\n";
+
+        mpDofUpdater = Kratos::make_unique<DofUpdaterType>(mRelaxationFactor);
+    }
+
+    AlgebraicFluxCorrectedSteadyScalarScheme(
+        const double RelaxationFactor,
+        const Flags BoundaryFlags,
+        const Variable<int>& rPeriodicIdVar)
+    : BaseType(),
+        mRelaxationFactor(RelaxationFactor),
+        mBoundaryFlags(BoundaryFlags),
+        mrPeriodicIdVar(rPeriodicIdVar)
+    {
+        KRATOS_INFO("AlgebraicFluxCorrectedSteadyScalarScheme")
+            << " Using periodic residual based algebraic flux corrected scheme "
+               "with relaxation "
+               "factor = "
+            << std::scientific << mRelaxationFactor << "\n";
+
+        mpDofUpdater = Kratos::make_unique<DofUpdaterType>(mRelaxationFactor);
+    }
+
+    ~AlgebraicFluxCorrectedSteadyScalarScheme() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    void Initialize(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY
+
+        BaseType::Initialize(rModelPart);
+
+        if (mrPeriodicIdVar != Variable<int>::StaticObject()) {
+            block_for_each(rModelPart.Conditions(), [&](const ModelPart::ConditionType& rCondition) {
+                if (rCondition.Is(PERIODIC)) {
+                    // this only supports 2 noded periodic conditions
+                    KRATOS_ERROR_IF(rCondition.GetGeometry().PointsNumber() != 2)
+                        << this->Info() << " only supports two noded periodic conditions. Found "
+                        << rCondition.Info() << " with "
+                        << rCondition.GetGeometry().PointsNumber() << " nodes.\n";
+
+                    const auto& r_node_0 = rCondition.GetGeometry()[0];
+                    const std::size_t r_node_0_pair_id =
+                        r_node_0.FastGetSolutionStepValue(mrPeriodicIdVar);
+
+                    const auto& r_node_1 = rCondition.GetGeometry()[1];
+                    const std::size_t r_node_1_pair_id =
+                        r_node_1.FastGetSolutionStepValue(mrPeriodicIdVar);
+
+                    KRATOS_ERROR_IF(r_node_0_pair_id != r_node_1.Id())
+                        << "Periodic condition pair id mismatch in "
+                        << mrPeriodicIdVar.Name() << ". [ " << r_node_0_pair_id
+                        << " != " << r_node_1.Id() << " ].\n";
+
+                    KRATOS_ERROR_IF(r_node_1_pair_id != r_node_0.Id())
+                        << "Periodic condition pair id mismatch in "
+                        << mrPeriodicIdVar.Name() << ". [ " << r_node_1_pair_id
+                        << " != " << r_node_0.Id() << " ].\n";
+                }
+            });
```

Comment:
I'd throw an error if the `if (mrPeriodicIdVar != Variable<int>::StaticObject())` is not satisfied (in any case, I don't think this can be technically achieved).

Reply:
A Ctor is used in the non-periodic case which initializes `mrPeriodicIdVar = Variable<int>::StaticObject()` as in the following line.
https://github.com/KratosMultiphysics/Kratos/blob/6ef39a640e77741da7c13244b91b64d0c95f9faf/applications/RANSApplication/custom_strategies/algebraic_flux_corrected_steady_scalar_scheme.h#L74-L80

Would it be ok to have it like that rather than havinga `bool` to check whther scheme is periodic or not?

Reply:
Understood.